pkg/services/postgresqldb/deprovision_db_only.go

```diff
@@ -0,0 +1,74 @@
+package postgresqldb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/open-service-broker-azure/pkg/service"
+)
+
+func (d *dbOnlyManager) GetDeprovisioner(
+	service.Plan,
+) (service.Deprovisioner, error) {
+	return service.NewDeprovisioner(
+		service.NewDeprovisioningStep("deleteARMDeployment", d.deleteARMDeployment),
+		service.NewDeprovisioningStep(
+			"deletePostgreSQLDatabase",
+			d.deletePostgreSQLDatabase,
+		),
+	)
+}
+
+func (d *dbOnlyManager) deleteARMDeployment(
+	_ context.Context,
+	instance service.Instance,
+) (service.InstanceDetails, error) {
+	dt, ok := instance.Details.(*dbOnlyPostgresqlInstanceDetails)
+	if !ok {
+		return nil, fmt.Errorf(
+			"error casting instance.Details as *dbOnlyPostgresqlInstanceDetails",
+		)
+	}
+	if err := d.armDeployer.Delete(
+		dt.ARMDeploymentName,
+		instance.ResourceGroup,
+	); err != nil {
+		return nil, fmt.Errorf("error deleting ARM deployment: %s", err)
+	}
+	return dt, nil
+}
+
+func (d *dbOnlyManager) deletePostgreSQLDatabase(
+	ctx context.Context,
+	instance service.Instance,
+) (service.InstanceDetails, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	pdt, ok := instance.Parent.Details.(*dbmsOnlyPostgresqlInstanceDetails)
+	if !ok {
+		return nil, fmt.Errorf(
+			"error casting instance.Parent.Details " +
+				"as *dbmsOnlyPostgresqlInstanceDetails",
+		)
+	}
+	dt, ok := instance.Details.(*dbOnlyPostgresqlInstanceDetails)
+	if !ok {
+		return nil, fmt.Errorf(
+			"error casting instance.Details " +
+				"as *dbOnlyPostgresqlInstanceDetails",
+		)
+	}
+	result, err := d.databasesClient.Delete(
+		ctx,
+		instance.Parent.ResourceGroup,
+		pdt.ServerName,
+		dt.DatabaseName,
+	)
```

Comment:
Does this work out well even though, after provisioning, we modified this DB some? (We change it to be owned by a certain role.) It probably is ok. Legit asking though because I haven't tried it.